test/core_organization_test.go

```diff
@@ -70,23 +70,38 @@ func TestCoreOrganization(t *testing.T) {
 	require.NoError(t, cl.Create(ctx, org))
 	require.NoError(t, testutil.WaitUntilReady(ctx, cl, org))
 
-	projectTemplate := &corev1alpha1.OrganizationRoleTemplate{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{
-		Name: templates.ProjectAdminOrganizationRoleTemplateName,
-	}, projectTemplate))
-
-	rbacTemplate := &corev1alpha1.OrganizationRoleTemplate{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{
-		Name: templates.RBACAdminOrganizationRoleTemplateName,
-	}, rbacTemplate))
+	projectTemplate := &corev1alpha1.OrganizationRoleTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: templates.ProjectAdminOrganizationRoleTemplateName,
+		},
+	}
+	require.NoError(t, testutil.WaitUntilReady(ctx, cl, projectTemplate))
+	rbacTemplate := &corev1alpha1.OrganizationRoleTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: templates.RBACAdminOrganizationRoleTemplateName,
+		},
+	}
+	require.NoError(t, testutil.WaitUntilReady(ctx, cl, rbacTemplate))
+	require.NoError(t, cl.WaitUntil(ctx, org, func() (done bool, err error) {
```

Comment:
You've already waited for organization being ready, no need for this again.

Reply:
This is needed, because organization will only propagate owner to member list if the role/rolebinding are created by OrganizationRoleTemplate.
The reason for this design is that: OrganizationRoleTemplate controller will only create roles/rolebindings in Organization which is ready(has a namespace created). So we need to mark Organization as ready and then create roles/rolebindings.

Reply:
We can only check if Organization namespace is created, as you have seen here: https://github.com/kubermatic/bulward/pull/31#discussion_r450127566
And then create role/rolebindings, and then check them in organization controller, then mark the organization as ready.
But I don't really see the benefits of this, and it is completely overcomplicated, so I would go with the current implementation and if we see it is an issue in the future, then we fix it.